tests/unit/generators/xml_files/test_create_configuration_xml.py

```diff
@@ -56,51 +39,31 @@ def test_generate_xml_without_configuration(
         addon_name=ta_name,
     )
 
-    mock_writer = MagicMock()
-    with patch.object(configuration_xml, "writer", mock_writer):
-        file_paths = configuration_xml.generate()
+    file_paths = configuration_xml.generate()
+    assert file_paths == {"": ""}
 
-        # Assert that no files are returned since no dashboard is configured
-        assert file_paths == {"": ""}
 
-
-@patch(
-    "splunk_add_on_ucc_framework.generators.xml_files.ConfigurationXml._set_attributes",
-    return_value=MagicMock(),
-)
-@patch(
-    "splunk_add_on_ucc_framework.generators.xml_files.ConfigurationXml.get_file_output_path"
-)
-def test_generate_xml(
-    mock_op_path,
-    mock_set_attributes,
-    global_config_all_json,
-    input_dir,
-    output_dir,
-    ucc_dir,
-    ta_name,
-):
+def test_generate_xml(global_config_all_json, input_dir, output_dir, ta_name):
     config_xml = ConfigurationXml(
         global_config_all_json,
         input_dir,
         output_dir,
-        ucc_dir=ucc_dir,
+        ucc_dir=UCC_DIR,
```

Comment:
Can we refactor `FileGenerator` class to not require `ucc_dir` as it will not change and will simplify writing tests and using those classes? Happy to see a separate PR for that.

Reply:
Without `ucc_dir`, we wouldn't be able to find the template files.
However, if we want to remove `ucc_dir`, I think we can absorb those templates into Python scripts, remove all template files and remove jinja2 dependency?

Reply:
The path to template files is static and should not be a parameter to `FileGenerator` class. There is only 1 value that can be accepted there so it is not a parameter and should not be exposed in the class itself.

Reply:
you can do - ```self._template_dir = [(sep.join([UCC_DIR, "templates"]))]``` in `FileGenerator` and remove all mentions of `ucc_dir` from the tests.

Reply:
Okay, got your point.